### ISSUE TITLE
Wordpress false positive is user password begins with "$"

### ISSUE THREAD

**Issue by dfumagalli** [coreruleset/wordpress-rule-exclusions-plugin#33]
Hello,

I had several issues adding new users in latest WordPress and latest OWASP WordPress rules exclusion plugin.

The log is as follows:

`[Sat Mar 02 11:48:20.631053 2024] [:error] [pid 27350] [client REDACTED:55579] [client REDACTED] ModSecurity: Warning. Pattern match "\\\\$+(?:[a-zA-Z_\\\\x7f-\\\\xff][a-zA-Z0-9_\\\\x7f-\\\\xff]*|\\\\s*{.+})(?:\\\\s|\\\\[.+\\\\]|{.+}|/\\\\*.*\\\\*/|//.*|#.*)*\\\\(.*\\\\)" at ARGS:pass1. [file "/usr/share/modsecurity-crs/rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf"] [line "460"] [id "933180
"] [msg "PHP Injection Attack: Variable Function Call Found"] [data "Matched Data: $mm82rEJpdrmYns(Oa) found within ARGS:pass1: $mm82rEJpdrmYns(Oa)889aV"] [severity "CRITICAL"] [ver "OWASP_CRS/3.3.2"] [tag "application-multi"] [tag "language-php"] [tag "platform-multi"] [tag "attack-injection-php"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/152/242"] [hostname "www.REDACTED.com"] [uri "/wp-admin/user-new.php"] [unique_id "ZeMSBJ3nkXCGEAmPg4frkwAAAAM
"], referer: https://www.REDACTED.com/wp-admin/user-new.php`

The matched data, found in ARGS:pass1, is exactly the random password WordPress created. I believe the pattern triggers because it thinks I am passing a PHP variable (they begin with "$").

**Comment by EsadCetiner** [comment]
@dfumagalli The false positive you reported should already be fixed by the plugin [here](https://github.com/coreruleset/wordpress-rule-exclusions-plugin/blob/a87d492c1caed2607ac125b12f5cdb600403ea08/plugins/wordpress-rule-exclusions-before.conf#L482). I've tried reproducing the issue myself and I was successfully able to create a new WordPress user with the password you provided.

I can see that you're using CRSv3.3.2, plugins are a new feature in CRSv4 and are not supported in 3.3.2 (however the WordPress plugin should work fine if you backported it to 3.3.2).

It's likely that you've installed the plugin incorrectly or that you have a 3rd party plugin installed in WordPress that changes how WordPress creates new users (3rd party WordPress plugins are currently not supported).

Can you provide some additional information about your environment such as what version of WordPress you're running, and what plugins you have installed (if any)?

**Comment by dfumagalli** [comment]
Hello,
I am using the rules I downloaded from this repository like 15 days ago.
I easily installed the plugin (I read the instructions on your main website).
All I had to do as a whole was to edit 1 rule (not relevant to here) and everything worked well in CRSv3.3.2.

Additional information: WordPress with *ZERO* plugins I downloaded today on Apache 2 + FPM / PHP 8.2 on Ubuntu.
Adding the user was literally the first action I did after installing WordPress.

**Comment by EsadCetiner** [comment]
@dfumagalli 
> WordPress with ZERO plugins I downloaded today 

So then I assume you're on WordPress 6.4.3 (I'm on 6.4.3 and this works fine for me)?

In that case, the issue is likely a config issue. I'm not very familiar with Apache so forgive me. How are you including the plugin within ModSecurity under ``/etc/apache2/mods-enabled/security2.conf``, are you loading it before CRS or after?

**Comment by dfumagalli** [comment]
`<IfModule security2_module>
        # Default Debian dir for modsecurity's persistent data
        SecDataDir /var/cache/modsecurity

        # Include all the *.conf files in /etc/modsecurity.
        # Keeping your local configuration in that directory
        # will allow for an easy upgrade of THIS file and
        # make your life easier
        IncludeOptional /etc/modsecurity/*.conf

        # Include OWASP ModSecurity CRS rules if installed
        IncludeOptional /usr/share/modsecurity-crs/*.load
</IfModule>`

**Comment by EsadCetiner** [comment]
@dfumagalli Thank You
I don't see any includes for the plugins directory here, that's probably your issue. I don't know where you placed the plugins file, but the ``security2.conf`` file should look something like this. Replace ``/path/to/coreruleset/plugins/`` with the path to where your plugin files are installed.

```
    # Default Debian dir for modsecurity's persistent data
    SecDataDir /var/cache/modsecurity

    # Include all the *.conf files in /etc/modsecurity.
    # Keeping your local configuration in that directory
    # will allow for an easy upgrade of THIS file and
    # make your life easier
    IncludeOptional /etc/modsecurity/*.conf

    Include /path/to/coreruleset/plugins/*-config.conf
    Include /path/to/coreruleset/plugins/*-before.conf

    # Include OWASP ModSecurity CRS rules if installed
    IncludeOptional /usr/share/modsecurity-crs/*.load
    
    Include /path/to/coreruleset/plugins/*-after.conf
```

**Comment by dfumagalli** [comment]
Ok, thank you!
